### PR TITLE
build(js): upgrade babel and webpack toolchains

### PR DESCRIPTION
**Pull request by mcous**
## overview

This PR upgrades the following portions of our JS build:

- Babel
    - Now with [nullish coalescing](https://github.com/tc39/proposal-nullish-coalescing/)!
    - Now with support for explicit inexact Flow objects (`{ ... }`)
- Webpack
    - Felt like a thing to do
    - Webpack v5 is currently in beta, so want to make sure we're up to date on v4 to minimize future upgrade pain
- React Hot Loader
    - Now with hooks support
- Jest
    - Because babel was upgraded

## review requests

Smoke tests require for all front end apps:

- [x] Protocol designer
- [x] App
- [x] Labware Library
- [x] Protocol Library Kludge (deckmap)

### CODE REVIEW

File: labware-library/package.json [hunk 1]
```diff
@@ -28,11 +29,8 @@
     "query-string": "^6.2.0",
     "react": "^16.8.6",
```
Comment (mcous) [inline labware-library/package.json]: TODO: lock react and react-dom

File: labware-library/package.json [hunk 2]
```diff
@@ -28,11 +29,8 @@
...
     "react-dom": "^16.8.6",
-    "react-hot-loader": "^4.8.0",
+    "react-hot-loader": "^4.12.19",
     "react-router-dom": "^5.0.1",
     "yup": "^0.27.0"
-  },
-  "devDependencies": {
-    "react-snap": "^1.23.0"
```
Comment (mcous) [inline labware-library/package.json]: All devDependencies go in monorepo package.json, where this entry already exists

File: package.json [hunk 3]
```diff
@@ -24,25 +24,22 @@
...
+    "@babel/plugin-proposal-class-properties": "^7.8.3",
+    "@babel/plugin-transform-modules-commonjs": "^7.8.3",
+    "@babel/preset-env": "^7.8.4",
+    "@babel/preset-flow": "^7.8.3",
+    "@babel/preset-react": "^7.8.3",
     "ajv": "^6.10.2",
     "aws-sdk": "^2.493.0",
-    "babel-eslint": "^10.0.3",
-    "babel-jest": "^24.9.0",
+    "babel-eslint": "^11.0.0-beta.2",
```
Comment (mcous) [inline package.json]: Beta release required for flow inexact object syntax; keep an eye on this

File: package.json [hunk 4]
```diff
@@ -24,25 +24,22 @@
...
+    "babel-jest": "^25.1.0",
```
Comment (mcous) [inline package.json]: Oops, probably shouldn't upgrade this without upgrading jest

File: package.json (hunk 3 above)
Comment (shlokamin) [reply package.json]: That's annoying — let's def keep an eye out

File: app/package.json [hunk 5]
```diff
@@ -32,9 +33,9 @@
     "moment": "^2.19.1",
     "netmask": "^1.0.6",
     "path-to-regexp": "^3.0.0",
-    "react": "^16.8.6",
-    "react-dom": "^16.8.6",
-    "react-hot-loader": "^3.0.0-beta.7",
+    "react": "16.8.6",
+    "react-dom": "16.8.6",
```
Comment (shlokamin) [inline app/package.json]: Just curious — why is it that we were previously using semantic versioning for react and not we're not?

File: components/package.json [hunk 6]
```diff
@@ -17,7 +17,7 @@
   "peerDependencies": {
     "classnames": "^2.2.5",
     "lodash": "^4.17.4",
-    "react": "^16.8.6",
+    "react": "16.8.6",
     "react-router-dom": "^5.0.1"
```
Comment (shlokamin) [inline components/package.json]: Same question about semantic versioning here

File: app/package.json (hunk 5 above)
Comment (mcous) [reply app/package.json]: Judgement call and a safety thing:

- In order for hooks support of react-hot-loader, we have to swap out `react-dom` with `@hot-loader/react-dom`
- The version of `@hot-loader/react-dom` needs to match `react-dom`
- We're already locking our dependencies implicitly via the lock file, so locking them down explicitly here felt like the safest thing to do

Longer term, I think we should probably lock all of our production dependencies explicitely (I'm generally good keeping dev deps on semver). That being said, I'd rather not mess with _all_ our app deps in this PR

**Comment by mcous** [comment]
Sorry about the image thing, forgot to read the changelog of `file-loader` after bumping it from v4 to v5. Pushed the short-term fix to the webpack config with the longer-term fix in a TODO comment

**Pull request merged**